backend/models/job.js

```diff
@@ -18,155 +18,205 @@
 "use strict";
 
 const {map, compact, uniq} = require("lodash");
-const mongoose = require("mongoose");
-const ModelFactory = require("./factory/modelFactory");
 const responseCodes = require("../response_codes.js");
 const C = require("../constants.js");
-const schema = mongoose.Schema({
-	_id: String,
-	color: String,
-	users: [String]
-});
+const db = require("../handler/db");
 
-function validateJobName(job) {
+function validateJobName(jobName) {
 	const regex = "^[^/?=#+]{0,119}[^/?=#+ ]{1}$";
-	return job && job.match(regex);
+	return jobName && jobName.match(regex);
 }
 
-schema.statics.addDefaultJobs = function(teamspace) {
+function getCollection(teamspace) {
+	return db.getCollection(teamspace, "jobs");
+}
+
+const Job = {};
+
+Job.addDefaultJobs = function(teamspace) {
 	const promises = [];
+
 	C.DEFAULT_JOBS.forEach(job => {
-		promises.push(this.addJob(teamspace, job));
+		promises.push(Job.addJob(teamspace, job));
 	});
 
 	return Promise.all(promises);
-
 };
 
-schema.statics.usersWithJob = function(teamspace) {
-	return this.find({account: teamspace}, {}, {_id: 1, users : 1}).then((jobs) => {
-		const userToJob  = {};
+Job.addJob = async function(teamspace, jobData) {
+	if (!jobData._id || !validateJobName(jobData._id)) {
+		throw responseCodes.JOB_ID_INVALID;
+	}
 
-		jobs.forEach(job => {
-			job.users.forEach(user => {
-				userToJob[user] = job._id;
-			});
-		});
+	const foundJob = await Job.findByJob(teamspace, jobData._id);
+
+	if (foundJob) {
+		throw responseCodes.DUP_JOB;
+	}
 
-		return userToJob;
+	const jobsColl = await getCollection(teamspace);
+	const newJobEntry = {
+		_id: jobData._id,
+		users: []
+	};
 
-	});
+	if (jobData.color) {
+		newJobEntry.color = jobData.color;
+	}
 
+	return jobsColl.insert(newJobEntry);
```

Comment:
I think it's a good idea to base this branch on the groups on so we have the changes on dbHandler to just call insert instead of doing awaits on collection

Reply:
Yes, I was hoping to update it once the other branch was merged